pandas/core/base.py

```diff
@@ -290,7 +291,9 @@ class SelectionMixin(object):
     }
 
     @property
-    def name(self):
+    def _selection_name(self):
+        """ return a name for myself; this would ideally be the 'name' property, but
+        we cannot conflict with the Series.name property which can be set """
```

Comment:
This explanation is not fully clear to me (but maybe I am not familiar enough with the groupby codebase)